lib/mix/test/mix/cli_test.exs

```diff
@@ -43,13 +43,16 @@ defmodule Mix.CLITest do
 
         def run(_) do
           IO.puts Mix.Project.get!.hello_world
+          Mix.shell.info("This won't appear")
         end
       end
       """
 
-      contents = mix ~w[my_hello]
+      contents = mix ~w[my_hello], [{"MIX_QUIET", "1"}]
+
       assert contents =~ "Hello from MyProject!\n"
-      assert contents =~ "Compiled lib/hello.ex\n"
+      refute contents =~ "Compiled lib/hello.ex\n"
```

Comment:
Ok @josevalim, changes done. My only doubt is that I had to change this line from `assert` to `refute` or this test won't pass (the `env` var changes the `Mix.shell`, so `"Compiled lib/hello.ex"` doesn't appear). I think that this doesn't change the validity of the test (compilation is tested in the previous tests), but I'll wait for your opinion. :+1: